### PR TITLE
Replacing line breaks with <br> in markdown text

### PR DESCRIPTION
**Pull request by janusbal**
# Description

Android, web and desktop support `\n` in markdown texts while iOS does not. This adds support to line breaks by replacing them with <br> html tag before generating the  html string.

# Sample Card
```
{
  "$schema": "http://adaptivecards.io/schemas/adaptive-card.json",
  "type": "AdaptiveCard",
  "version": "1.0",
  "body": [
 
    {
 
      "type": "Container",
 
      "items": [
 
        {
 
          "type": "TextBlock",
 
          "text": "Hello User",
 
          "wrap": true
 
        },
 
        {
 
          "type": "TextBlock",
 
          "text": "**bold**\n\n_cursive_",
 
          "wrap": true
 
        }
 
      ]
 
    }
 
  ]
}
```

# How Verified

How you verified the fix, including one or all of the following:
1. New unit tests that were added if any. If none were added please add a quick line explaining why not.
2. Existing relevant unit/regression tests that you ran
3. Manual scenario verification if any; ***Do include .gif's or screenshots of the testing you performed here if you think that it
will aid in code reviews or corresponding fixes on other platforms for eg.***

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/AdaptiveCards/pull/7184)

### CODE REVIEW

File: source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/UtiliOS.mm [hunk 1]
```diff
@@ -682,6 +682,11 @@ void buildIntermediateResultForText(ACRView *rootView, ACOHostConfig *hostConfig
     auto markdownString = markDownParser->TransformToHtml();
     NSString *parsedString = (markDownParser->HasHtmlTags()) ? [NSString stringWithCString:markdownString.c_str() encoding:NSUTF8StringEncoding] : [NSString stringWithCString:markDownParser->GetRawText().c_str() encoding:NSUTF8StringEncoding];
 
+    if (markDownParser->HasHtmlTags() && ([parsedString containsString:@"\n"] || [parsedString containsString:@"\r"])) {
```
Comment (jwoo-msft) [inline source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/UtiliOS.mm:685]: you can do something like this.
    parsedString = [parsedString stringByReplacingOccurrencesOfString:@"[\\n\\r]" withString:@"<br>" options:NSRegularExpressionSearch range:NSMakeRange(0, [parsedString length])];

Comment (janusbal) [reply source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/UtiliOS.mm:685]: good point, thanks!

**Review by jwoo-msft — APPROVED** [review]
@janusbal 
 
I think you forgot to run clang-format.

other than that, it looks good.

File: source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/UtiliOS.mm [hunk 2]
```diff
@@ -682,6 +682,12 @@ void buildIntermediateResultForText(ACRView *rootView, ACOHostConfig *hostConfig
...
```
Comment (licanhua) [inline source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/UtiliOS.mm:685]: Do we support complex syntax like this? If it's not handled carefully, possible you add wrong <br> into the output.
 

image

Comment (jwoo-msft) [reply source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/UtiliOS.mm:685]: that syntax is handled at the MarkDown parser in shared model.
His PR is handling line breaks after the text is parsed by the MarkDown parser.

**Review by licanhua — APPROVED** [review]

**Pull request merged**